### DISCUSSION TITLE
The model is at the last in a pipeline

### DISCUSSION THREAD

**Discussion by SwamiPatil** [Trusted-AI/adversarial-robustness-toolbox#1497] [Q&A]
https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/79a980f078b791e462ff47744836db913b590d0b/art/estimators/scikitlearn.py#L50
Current value of _input_shape is:
elif hasattr(model, "steps"):
_input_shape = self._get_input_shape(model.steps[0][1])

It should be 
_input_shape = self._get_input_shape(model.steps[-1][-1])

Because the model is at the last in pipeline.steps

**Comment by beat-buesser** [comment]
Hi @SwamiPatil Thank you very much for using ART! The `_input_shape` as return value for the `input_shape` property is expected to return the shape of the input into the model, in the case of a Pipeline the pipeline is treated as the model (although containing another model) for the purpose of the sklearn estimator.

Did you encounter a case where this causes a problem?
